File: data/multilang_comparison/program_julia/src/func50.jl
include("program_julia_head.jl")
function func50(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array24 = nothing
   if pCounter > 1
       array24 = vars.data[pCounter]
       pCounter -= 1
       array24.refC += 1
   else
       array24 = JArray(zeros(Int, 399), 399, 1, 24)
   end
   for i in 0:(array24.size-1) 
      array24.data[i+1] -= 1
   end
   for i in 1:array24.size
       if array24.data[i] == 76
               return array24
      end
   end
   return array24;
end

File: data/multilang_comparison/program_c/src/func50.c
#include "program_c.h" 
array_t* func50(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array24;
   if (pCounter > 0) {
      array24 = vars->data[--pCounter];
      array24->refC++;
      DEBUG_COPY(array24->id);
   } else {
      array24 = (array_t*)malloc(sizeof(array_t));
      array24->size = 399;
      array24->refC = 1;
      array24->id = 24;
      array24->data = (unsigned int*)malloc(array24->size*sizeof(unsigned int));
      memset(array24->data, 0, array24->size*sizeof(unsigned int));
      DEBUG_NEW(array24->id);
   }
   for (int i = 0; i < array24->size; i++) {
      array24->data[i]--;
   }
   for (int i = 0; i < array24->size; i++) {
      if (array24->data[i] == 76) { 
         return array24;
      }
   }
   return array24;
}

